### ISSUE TITLE
trouble with recreating the result with pretrained model

### ISSUE THREAD

**Issue by Delavari-Alireza** [arplaboratory/STHN#13]
hello, 
great work!
I have tried to recreate the result using the pretrained model provided in the google drive and the dataset introduced in this paper but the I get wrong results. what am I doing wrong? I need just the STHN.pth to run the model right?

I attach some of my result. the bounding box is always on top right corner of the images.

looking forward to hear from you soon.
best regards
![train_overlap_bbox_9](https://github.com/user-attachments/assets/3a8e4c5b-a23e-4565-8690-8c68d241b11c)

**Comment by xjh19971** [comment]
Thank you for exploring our work! It seems like this might be a visualization issue, potentially caused by incorrect axis limits or resizing settings.

* Axis Limits: Could you try reducing the axis limits to see if it resolves the issue?
* Resizing Settings: If the issue persists, could you debug by outputting the dimension information of the visualized results?

Additionally, could you let me know which specific line of code you use for the visualization? I’d be happy to assist further. Thank you!

**Comment by Delavari-Alireza** [comment]
Thank you for your reply! I truly appreciate your guidance. After some debugging, I found that resizing during visualization resolved the issue. However, I noticed some inconsistencies in the interface and code, which I would like to point out below.

adding resize to **save_overlap_bbox_img**
```python
def save_overlap_bbox_img(img1, img2, path, four_point_gt, four_point_pred, crop_bbox=None):
    four_point_gt = np.round(four_point_gt.cpu().numpy())
    four_point_pred = np.round(four_point_pred.cpu().numpy())
    plt.figure(figsize=(50, 10), dpi=200)
    plt.axis('off')

    img1_list = np.empty((img1.shape[0], img1.shape[2], img1.shape[3], img1.shape[1]))
    img2_list = np.empty((img1.shape[0], img1.shape[2], img1.shape[3], img1.shape[1]))

    for i in range(len(img2)):
        image1 = inv_base_transforms(img1[i].detach().cpu())
        image1 = np.array(image1)

        image2 = inv_base_transforms(img2[i].detach().cpu())
        image2 = np.array(image2)

        four_point_gt_single = np.int32(four_point_gt[i]).reshape((-1, 1, 2))
        four_point_pred_single = np.int32(four_point_pred[i]).reshape((-1, 1, 2))

        # Swap coordinates to match visualization needs
        four_point_gt_single[[2, 3]] = four_point_gt_single[[3, 2]]
        four_point_pred_single[[2, 3]] = four_point_pred_single[[3, 2]]

        image2 = cv2.polylines(image2, [four_point_gt_single], True, (0, 255, 0), 2)
        image2 = cv2.polylines(image2, [four_point_pred_single], True, (255, 0, 0), 1)

        if crop_bbox is not None:
            crop_bbox_single = np.int32(crop_bbox[i]).reshape((-1, 1, 2))
            crop_bbox_single[[2, 3]] = crop_bbox_single[[3, 2]]
            image2 = cv2.polylines(image2, [crop_bbox_single], True, (0, 0, 255), 1)

        image2 = cv2.resize(image2, (image1.shape[0], image1.shape[1]))
        img1_list[i] = image1
        img2_list[i] = image2

    img1_tensor = torch.from_numpy(img1_list).permute(0, 3, 1, 2)
    img2_tensor = torch.from_numpy(img2_list).permute(0, 3, 1, 2)
    img1_tensor = torchvision.utils.make_grid(img1_tensor, nrow=16, padding=0, pad_value=255)
    img2_tensor = torchvision.utils.make_grid(img2_tensor, nrow=16, padding=0, pad_value=255)

    img1 = np.array(img1_tensor.permute(1, 2, 0)).astype(np.uint8)
    img2 = np.array(img2_tensor.permute(1, 2, 0)).astype(np.uint8)

    plt.imshow(img2)
    plt.imshow(img1, alpha=0.25)
    plt.savefig(path, bbox_inches="tight", pad_inches=0)
    plt.close()
```

I also made changes to the following lines in **myevaluate.py**:

Updated the unpacking of data_blob:

```python
# Original
# img1, img2, flow_gt, H, query_utm, database_utm, image1_ori, index, pos_index = [x for x in data_blob]
# Modified
img1, img2, flow_gt, H, query_utm, database_utm, index, pos_index = [x for x in data_blob]
```
Updated the model forward call:

```python
# Original
# model.forward(use_raw_input=(args.train_ue_method == 'train_only_ue_raw_input'), noise_std=args.noise_std, sample_method=args.sample_method)
# Modified
model.forward()
```


### **Additional Question**
I also wanted to ask if the model can be applied to images where the real-world dimensions are unknown. Specifically, can the query image have a size other than 256m × 256m? If so, could you explain how to adapt the model for such cases?

Please let me know if you have further suggestions. Thank you again for your help!

Best regards,
Alireza Delavari

![Image](https://github.com/user-attachments/assets/30aff043-bbe6-4a3f-abd9-d8b501896efa)

![Image](https://github.com/user-attachments/assets/90c393d8-ae1f-4d1a-b2e7-6e8ec58525ab)

**Comment by xjh19971** [comment]
Thank you for your comments. I have incorporated the suggested changes into the main branch.

Regarding your question, our reasoning is that we aim to avoid resizing the satellite images to dimensions that are too small. Through experimentation, we found that resizing from 1536x1536 to 256x256 for satellite images provides an optimal scaling ratio of approximately 6x smaller. To maintain consistency, we also resize the thermal images to 256x256. This 256x256 resolution is sufficient for computational efficiency.

That said, you are free to use a different original input size and resized image size, as long as the original image is not resized to dimensions that are too small, and the resized image size is not too large to impact inference efficiency.

**Comment by xjh19971** [comment]
Close this issue if there are no follow-up questions
